desktop/apps/auction2/components/page/Footer.js

```diff
@@ -0,0 +1,66 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import block from 'bem-cn'
+import buildTemplateComponent from 'desktop/components/react/utils/build_template_component'
+import { first } from 'underscore'
+
+export default function Footer (props) {
+  const { articles, auction, footerItems } = props
+  const footerItem = first(footerItems)
+  const showArticles = Boolean(articles.length)
+  const showFooterItems = footerItem && !auction.isAuctionPromo()
+  const b = block('auction-footer')
+
+  return (
+    <footer
+      className={b('')
+        .mix('auction-page-section')
+        .mix(articles.length ? 'has-articles' : 'has-no-articles')}
+    >
+      { showArticles &&
+        <div className={b('auction-articles')}>
+          { articles.models.map((article, key) => {
+            const ArticleFigure = buildTemplateComponent('desktop/components/article_figure/template.jade', { locals: { article } })
```

Comment:
Bits like this are useful for when porting over an app to react. While you don't have a react component you can render jade templates with this util, on the server. On the client it will throw so the idea is if something is too complex cases will be handled on a case-by-case basis:

```javascript
const TemplateComponent = isServer 
  ? buildTemplateComponent('path/to.jade', props)
  : () => <div>require('path/to.jade')(props)</div>

<div>
  <TemplateComponent />   
</div>
```

And so on.